documentation/modules/deploying/proc-deploy-topic-operator-standalone.adoc

```diff
@@ -144,6 +144,28 @@ env:
 <3> The keystore contains the private key for mTLS authentication.
 <4> The password for accessing the keystore.
 
+. If you need a custom `SASL` configuration. For example, to access a cluster in a cloud provider with a custom login module, the `custom` `SASL_MECHANISM` allows for `SASL_CUSTOM_CONFIG`. +
+*NOTE:  Only SASL properties can be set with the `custom` mechanism.*
++
+.Example custom SASL configuration
+[source,shell,subs=+quotes]
+----
+# ....
+env:
+  - name: SASL_MECHANISM # <1>
+    value: custom
+  - name: SASL_CUSTOM_CONFIG # <2>
```

Comment:
1. I wonder if `SASL_CUSTOM_CONFIG_JSON` would be clearer. 
2. But also, did you consider using `Properties` format rather than JSON? Users would likely be familiar with this because it's what the Kafka Java client's use, so it might avoid them having to convert something they already have as `Properties` format into JSON.

Reply:
1. I like adding JSON to the name.
2. JSONs easy to parse and I was worried about how it could be a mess getting people to understand the value as properties, but taking properties shouldn't be a big change if you would prefer that.

Reply:
> I was worried about how it could be a mess getting people to understand the value as properties, but taking properties shouldn't be a big change if you would prefer that.

That's a good point. The case of accidentally using folded style block strings (`>`), like this, bothers me

```yaml
- name: SASL_CUSTOM_CONFIG_PROPERTIES
  value: >
    sasl.mechanism: AWS_MSK_IAM
    sasl.jaas.config: software.amazon.msk.auth.iam.IAMLoginModule required;
    sasl.client.callback.handler.class: software.amazon.msk.auth.iam.IAMClientCallbackHandler
```

That's legal YAML and legal `.properties` format, but it's not what the user intended because it's just the single key `sasl.mechanism` with a malformed value made up of the real value and the remaining keys and values. Kafka itself would catch that when/if there was sufficient validation on the first property defined. I think there's a similar issue with plain flow scalar strings. (I guess this kind of problem is worse than other common YAML problems, like writing `foo: true` [where the value is expected to be a string not a boolean, so needs to be quoted] precisely because in this case the expected and actual types are the same, so it's less likely to be caught.)

So I agree, on balance, that although the JSON is ugly it's less likely to be parsed wrongly. 

Can we though:
* Call it `SASL_CUSTOM_CONFIG_JSON`
* Configure the `ObjectMapper` to support comments. It's non-standard, but I think people might appreciate the possibility to comment on why/how they're setting individual properties.

Reply:
I'm on it.

Reply:
This is what I'm working on now. Is `SASL_CUSTOM_CONFIG_JSON` and not having a custom type the solution we are going with?

Reply:
@john-mcpeek yes, I think that's what we landed on, right @fvaleri ?